internal/handlers/common.go

```diff
@@ -1,19 +1,212 @@
+	photoDeleted   = "photo deleted"
+
+	// Registration Confirm
+
+	errRegistrationAlreadyConfirmed = "Registration is already confirmed."
+	errRegistrationCannotFindUser   = "Can't find this user, registration confirmation link wasn't send"
+	errRegistrationCannotSend       = "Can't send registration confirmation link. Please try again later"
+	errFailedToConfirm              = "Failed to verify confirmation token. Please try again later"
+	confirmationNotRequired         = "Confirmation link was not sent to email, sending parameter was set to false and not required"
+	confirmationSent                = "Confirmation link was sent"
+	registrationConfirmed           = "You have successfully confirmed registration"
+
+	// Roles
+
+	errQueryRoles = "can't get all roles"
+
+	// Subcategory
+
+	errCreateSubcategory   = "failed to create new subcategory"
+	errMapSubcategory      = "failed to map new subcategory"
+	errQuerySCatByCategory = "failed to list subcategories by category id"
+	errGetSubcategory      = "failed to get subcategory"
+	errDeleteSubcategory   = "failed to delete subcategory"
+	errUpdateSubcategory   = "failed to update subcategory"
+	subcategoryDeleted     = "subcategory deleted"
+
+	// User
+
+	errInvalidLoginOrPass   = "invalid login or password"
+	errLoginInUse           = "login is already used"
+	errCreateUser           = "failed to create user"
+	errInvalidToken         = "token invalid"
+	errTokenRefresh         = "error while refreshing token"
+	errMapUser              = "map user error"
+	errUserNotFound         = "can't find user by id"
+	errUpdateUser           = "can't update user"
+	errRoleRequired         = "role id is required"
+	errSetUserRole          = "set user role error"
+	errQueryTotalUsers      = "failed get user total amount"
+	errQueryUsers           = "failed to get user list"
+	errDeleteUser           = "can't delete user"
+	errDeleteUserNotRO      = "user must be readonly for deletion"
+	errUserPasswordChange   = "error while changing password"
+	errWrongPassword        = "wrong password"
+	errPasswordsAreSame     = "old and new passwords are the same"
+	errPasswordPatchEmpty   = "password patch is empty"
+	errUpdateROAccess       = "error while updating readonly access"
+	errChangeEmail          = "error while changing email"
+	errEmailPatchEmpty      = "email patch is empty"
+	errNewEmailConfirmation = "can't send link for confirmation new email"
+	logoutSuccessful        = "successfully logged out"
+	roleAssigned            = "role assigned"
+)
+
+func buildErrorPayload(code int32, msg string, details string) *models.SwaggerError {
+	return &models.SwaggerError{
+		Code:    &code,
+		Message: &msg,
+		Details: details, // optional field for raw err messages
 	}
 }
 
-func buildStringPayload(msg string) *models.Error {
-	return &models.Error{
-		Data: &models.ErrorData{
-			Message: msg,
-		},
-	}
+func buildInternalErrorPayload(msg string, details string) *models.SwaggerError {
```

Comment:
Probably, `code` field doesn't make any sense for FE, for all our errors this code is equal to HTTP code and we can replace all this functions by 1 and use single code, for ex, 600.